src/main/java/org/mondemand/Client.java

```diff
@@ -54,6 +56,8 @@
   private static final boolean DEFAULT_AUTO_EMIT = false;   // auto emit disabled by default
   private static final boolean DEFAULT_CLEAR_STAT = false;  // clear stats after flush by auto emit
 
+  private static final Pattern keyPattern = Pattern.compile("[\\w\\.-]+");     // valid values for key: a-z A-Z 0-9 _ - .
```

Comment:
I asked Tim for the valid characters for mondemand keys, but can someone confirm this?

Reply:
This is fine although the graphite part of the rrd backend will normalize this further by converting any characters not in [a-zA-Z0-9] to '_'.  So leaving '-' and '.' is fine but be aware those *may* be removed by some backends.

Reply:
Tim mentioned that they are getting a lot of stats with keys that are invalid, that is why I added this.